internal/interfaces/controller/azuredevops_controller.go

```
@@ -155,6 +154,7 @@
 	// Add finalizer if it doesn't exist
 	if !controllerutil.ContainsFinalizer(&cr, azuredevops.AzdoFinalizerName) {
 		controllerutil.AddFinalizer(&cr, azuredevops.AzdoFinalizerName)
+		r.Update(ctx, &cr)
```

golangci-lint (1.64.8)

Error return value of `r.Update` is not checked

(errcheck)